packages/@blockprotocol/graph/src/stdlib/subgraph/roots.ts

```diff
@@ -0,0 +1,151 @@
+import { isEntityEditionId } from "../../types/entity.js";
+import { isOntologyTypeEditionId } from "../../types/ontology.js";
+import {
+  Subgraph,
+  SubgraphRootType,
+  SubgraphRootTypes,
+} from "../../types/subgraph.js";
+import { Vertex } from "../../types/subgraph/vertices.js";
+import { mustBeDefined } from "../must-be-defined.js";
+import { getDataTypeByEditionId } from "./element/data-type.js";
+import { getEntity } from "./element/entity.js";
+import { getEntityTypeByEditionId } from "./element/entity-type.js";
+import { getPropertyTypeByEditionId } from "./element/property-type.js";
+
+/**
+ * Returns all root elements.
+ *
+ * The type of this can be constrained by using some of the helper type-guards:
```

Comment:
I think this is a bit clearer, if I've understood the intent
```suggestion
 * For a narrower return type, first narrow the type of `subgraph` by using one of the helper type-guards:
```

Reply:
Good call, thanks

[Improve doc](https://github.com/blockprotocol/blockprotocol/pull/820/commits/bf123b4e52437d7ec340ef5b0211175417983ae1)